includes/Media.php

```diff
@@ -273,4 +277,42 @@ public static function upload_mimes( array $mime_types ) {
 		$mime_types['svg'] = 'image/svg+xml';
 		return $mime_types;
 	}
+
+	/**
+	 * Deletes associated poster image when a video is deleted.
+	 *
+	 * This prevents the poster image from becoming an orphan because it is not
+	 * displayed anywhere in WordPress or the story editor.
+	 *
+	 * @param int $attachment_id ID of the attachment to be deleted.
+	 *
+	 * @return void
+	 */
+	public static function delete_video_poster( $attachment_id ) {
+		remove_action( 'pre_get_posts', [ __CLASS__, 'filter_poster_attachments' ] );
+		$query = new WP_Query(
+			[
+				'fields'         => 'ids',
+				'post_status'    => 'any',
+				'post_type'      => 'attachment',
+				'post_parent'    => $attachment_id,
+				'no_found_rows'  => true,
+				'posts_per_page' => 10,
```

Comment:
Should we add `ignore_sticky_posts`, `update_post_term_cache` here as well?

Reply:
Yep, makes sense 👍